ksqldb-engine/src/main/java/io/confluent/ksql/topic/TopicCreateInjector.java

```diff
@@ -175,13 +183,12 @@ public TopicCreateInjector(
 
   private TopicProperties createTopic(
       final Builder topicPropertiesBuilder,
-      final boolean shouldCompactTopic
+      final String topicCleanUpPolicy
   ) {
     final TopicProperties info = topicPropertiesBuilder.build();
 
-    final Map<String, ?> config = shouldCompactTopic
-        ? ImmutableMap.of(TopicConfig.CLEANUP_POLICY_CONFIG, TopicConfig.CLEANUP_POLICY_COMPACT)
-        : Collections.emptyMap();
+    final Map<String, ?> config =
+        ImmutableMap.of(TopicConfig.CLEANUP_POLICY_CONFIG, topicCleanUpPolicy);
```

Comment:
it seems that for create stream previously we were using the default for the cluster, but now we've changed that to explicitly specify delete. Is that correct/desired?

Reply:
The default is delete: https://kafka.apache.org/documentation/#cleanup.policy

I thought explicitly specifying the policy was cleaner than having empty vs non-empty maps, but if there are situations you're concerned about in terms of correctness we should definitely investigate / return to passing empty maps.

Reply:
Something to check is to see if the `CREATE STREAM` code path will change the config of a topic to delete if it already exists. Another thing to consider if it's possible (though I wouldn't understand why anyone would do this) to change the default for the cluster for cleanup.policy. If neither of these are a real concern, then I'm happy making it explicit

Reply:
Turns out it's possible to change the default cleanup.policy using the broker config log.cleanup.policy. I've switched this PR back to specifying empty configs in the case of streams in order to avoid a breaking change, but this behavior (using the default for streams and forcing tables to be compacted) feels weird to me. Is it preferable to use the broker default for streams, rather than specifying delete?

Reply:
Good question, there's definitely a strong argument toward forcing delete for streams... I could be convinced either way, perhaps now I'm actually leaning a little bit toward your original code.

Reply:
OK, I'll go with the original then.

Regarding your other question:

> Something to check is to see if the CREATE STREAM code path will change the config of a topic to delete if it already exists.

This never happens since the KafkaTopicClient does not update topics if they exist. (It also doesn't even validate the cleanup policy, only number of replicas and partitions: https://github.com/confluentinc/ksql/blob/220f7aaa97bd4a53467b3f4da4a46941e63ae933/ksqldb-engine/src/main/java/io/confluent/ksql/services/KafkaTopicClientImpl.java#L96)